tests/testthat/test_base_calculateConfusionMatrix.R

```diff
@@ -118,3 +118,38 @@ test_that("calculateConfusionMatrix with different factor levels (#2030)", {
   expect_equal(cm$result[1, 4], 0)
   expect_equal(cm$result[4, 4], 5)
 })
+
+
+test_that("calculateConfusionMatrix set argument works", {
+    mod = train("classif.lda", iris.task)
+    pred1 = predict(mod, iris.task)
+    rdesc = makeResampleDesc("CV", iters = 10, predict = "both")
+    # here, you have set=train and set=test in pred3$data:
+    pred2 = resample("classif.rpart", iris.task, rdesc)$pred
+
+    # pred1$data has no column "set" => argument set="train" would *not* make sense
+    expect_error(calculateConfusionMatrix(pred1, set = "train"))
+
+    # pred3 was predicted on both train and test set. Both subsetted matrices should give
+    # a positive total count:
+    expect_gt(sum(calculateConfusionMatrix(pred2, set = "train")$result), 0)
+    expect_gt(sum(calculateConfusionMatrix(pred2, set = "test")$result), 0)
```

Comment:
Could you test the actual numbers please, not just greater than 0? If the set argument did nothing and just returned everything all the time this would still pass.

There are some smaller test tasks in tests/testthat/helper_objects.R that you can use to speed things up and make checking the result easier.

Reply:
Okay, done. The lines look a bit messy now, I subsetted the confusion matrices to [1:3]. I hope that's still okay.